src/lib/get-raw-block.ts

```diff
@@ -1,133 +1,19 @@
-    console.error('cause', cause)
-    throw new Error(`unable to fetch raw block for CID ${cid}`)
-  }
-}
+import { BlockFetchTimeoutError } from './errors.js'
 
 /**
- * This method validates that the block we got from the gateway has the same CID as the one we requested
+ * Method for getting a raw block either with helia from trustless gateways or a local Kubo gateway.
  */
-export async function verifyBytes (providedCid: CID, bytes: Uint8Array): Promise<void> {
-  try {
-    const cid = await getCidFromBytes(bytes, providedCid.version, providedCid.code, providedCid.multihash.code)
-
-    if (cid.toString() !== providedCid.toString()) {
-      throw new Error(`CID mismatch, expected '${providedCid.toString()}' but got '${cid.toString()}'`)
-    }
-  } catch (err) {
-    console.error('unable to verify bytes', err)
-    throw err
-  }
-}
-
-/**
- * allow users to disable fetching from gateways by setting 'explore.ipld.gatewayEnabled' to false in localStorage
- *
- * @returns {boolean}
- */
-function ensureGatewayFetchEnabled (): boolean {
-  console.info('import.meta.env.NODE_ENV: ', import.meta.env.NODE_ENV)
-  console.info(
-    "🎛️ Customise whether ipld-explorer-components fetches content from gateways by setting an `explore.ipld.gatewayEnabled` value to true/false in localStorage. e.g. localStorage.setItem('explore.ipld.gatewayEnabled', false) -- NOTE: defaults to true"
-  )
-  const gatewayEnabledSetting = localStorage.getItem('explore.ipld.gatewayEnabled')
-
-  return gatewayEnabledSetting != null ? JSON.parse(gatewayEnabledSetting) : true
-}
-
-export async function getBlockFromAnyGateway (cid: CID, signal: AbortSignal, moreGateways: string[] = []): Promise<Uint8Array> {
-  if (!ensureGatewayFetchEnabled()) {
-    throw new Error('Fetching from gateways is disabled')
-  }
-  const gateways = moreGateways.concat(defaultGateways)
-  for (const url of gateways) {
-    if (signal.aborted) {
-      throw new Error('aborted')
-    }
-    try {
-      const rawBlock = await getRawBlockFromGateway(url, cid, signal)
-      try {
-        await verifyBytes(cid, rawBlock)
-        return rawBlock
-      } catch (err) {
-        console.error('unable to verify block from gateway', url)
-        continue
-      }
-    } catch (err) {
-      console.error('unable to get block from gateway', err)
-      // ignore the error
-    }
-  }
-  throw new Error('Could not get block from any gateway')
-}
-
-const defaultGateways = ['https://ipfs.io', 'https://dweb.link']
-
-/**
- * Method for getting a raw block either with helia or https://docs.ipfs.tech/reference/http/gateway/#trusted-vs-trustless
- * inspiration from https://github.com/ipfs-shipyard/ipfs-geoip/blob/466cd9d6454098c0fcf998b2217225099a654695/src/lookup.js#L18
- */
-export async function getRawBlock (helia: Helia, kuboClient: IPFSHTTPClient, cid: CID, timeout = 30000): Promise<Uint8Array> {
+export async function getRawBlock (helia: Helia, cid: CID, timeout = 30000): Promise<Uint8Array> {
```

Comment:
This is very close to where the timeout is used and is override-able via code.. I think this is fine for now to prevent the need to refactor a lot more